src/sql/workbench/contrib/query/browser/query.contribution.ts

```diff
@@ -522,6 +522,11 @@ const registryProperties: { [path: string]: IConfigurationPropertySchema; } = {
 		'type': 'boolean',
 		'default': true,
 		'description': localize('mssql.query.ansiNulls', "Enable SET ANSI_NULLS")
+	},
+	'mssql.query.alwaysEncryptedParameterization': {
```

Comment:
So if a user runs a query that fails because it isn't parameterized what error do they get back? Can we check for that and prompt them to enable this setting when that happens?

Reply:
The returned error is the following:

`Msg 33299, Level 16, State 6, Line 2
Encryption scheme mismatch for columns/variables '@SSN'. The encryption scheme for the columns/variables is (encryption_type = 'PLAINTEXT') and the expression near line '2' expects it to be (encryption_type = 'DETERMINISTIC', encryption_algorithm_name = 'AEAD_AES_256_CBC_HMAC_SHA_256', column_encryption_key_name = 'CEK_Auto1', column_encryption_key_database_name = 'Clinic') (or weaker).`

This Encryption scheme mismatch can be raised in a few of situations other than when parameterization is not enabled so I do not think this would be a good trigger for such a suggestion, however, in SSMS we trigger such a suggestion when a user is both connected with Always Encrypted enabled and when a query is executed. We could likely do the same here. I will mention this to the team and create a work item.

![image](https://user-images.githubusercontent.com/4079568/81101863-95ba4180-8ec3-11ea-95bf-86a592b2b9e8.png)